src/System.Private.DataContractSerialization/src/Resources/System.Private.DataContractSerialization.rd.xml

```diff
@@ -10,6 +10,45 @@
           </Method>
         </Type>
         <Type Name="KeyValuePairAdapter`2" Dynamic="Required All" />
+        <!-- Reflection-based serialization requires the entries below. -->
+        <Type Name="CollectionDataContract">
+          <Type Name="CollectionDataContractCriticalHelper">
+            <Method Name="BuildCreateGenericDictionaryEnumerator{K, V}" Dynamic="Required">
+              <GenericParameter Name="K" Dynamic="Required All" />
```

Comment:
Are these generic parameters and the others in the file expected to be user types or internal types? If they're internal types, this change is fine. If they're user types, I'd both be concerned this won't work and suspect the directives may need to be fine-tuned for size impact.